sdk/core/azure-core-serializer-json-gson/src/main/java/com/azure/core/serializer/json/gson/GsonJsonPrimitive.java

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.serializer.json.gson;
+
+import com.azure.core.util.serializer.JsonPrimitive;
+
+import java.util.Objects;
+
+/**
+ * GSON specific implementation of {@link JsonPrimitive}.
+ */
+public final class GsonJsonPrimitive implements JsonPrimitive {
+    private final com.google.gson.JsonPrimitive jsonPrimitive;
+
+    /**
+     * Constructs a {@link JsonPrimitive} wrapping the passed boolean.
+     *
+     * @param value Boolean value to wrap.
+     */
+    public GsonJsonPrimitive(boolean value) {
+        this.jsonPrimitive = new com.google.gson.JsonPrimitive(value);
+    }
+
+    /**
+     * Constructs a {@link JsonPrimitive} wrapping the passed double.
+     *
+     * @param value Double value to wrap.
+     */
+    public GsonJsonPrimitive(double value) {
+        this((Number) value);
+    }
+
+    /**
+     * Constructs a {@link JsonPrimitive} wrapping the passed float.
+     *
+     * @param value Float value to wrap.
+     */
+    public GsonJsonPrimitive(float value) {
+        this((Number) value);
+    }
+
+    /**
+     * Constructs a {@link JsonPrimitive} wrapping the passed int.
+     *
+     * @param value Int value to wrap.
+     */
+    public GsonJsonPrimitive(int value) {
+        this((Number) value);
+    }
+
+    /**
+     * Constructs a {@link JsonPrimitive} wrapping the passed long.
+     *
+     * @param value Long value to wrap.
+     */
+    public GsonJsonPrimitive(long value) {
+        this((Number) value);
+    }
+
+    /**
+     * Constructs a {@link JsonPrimitive} wrapping the passed short.
+     *
+     * @param value Short value to wrap.
+     */
+    public GsonJsonPrimitive(short value) {
+        this((Number) value);
+    }
+
+    private GsonJsonPrimitive(Number number) {
+        this.jsonPrimitive = new com.google.gson.JsonPrimitive(number);
+    }
+
+    /**
+     * Constructs a {@link JsonPrimitive} wrapping the passed string.
+     *
+     * @param value String value to wrap.
+     */
+    public GsonJsonPrimitive(String value) {
+        this.jsonPrimitive = new com.google.gson.JsonPrimitive(value);
+    }
+
+    /**
+     * Constructs a {@link JsonPrimitive} backed by the passed GSON {@link com.google.gson.JsonPrimitive}.
+     *
+     * @param jsonPrimitive The backing GSON {@link com.google.gson.JsonPrimitive}.
+     */
+    public GsonJsonPrimitive(com.google.gson.JsonPrimitive jsonPrimitive) {
+        this.jsonPrimitive = jsonPrimitive;
+    }
+
+    com.google.gson.JsonPrimitive getJsonPrimitive() {
+        return jsonPrimitive;
+    }
+
+    @Override
+    public boolean isBoolean() {
+        return jsonPrimitive.isBoolean();
+    }
+
+    @Override
+    public boolean getBoolean() {
+        return jsonPrimitive.isBoolean();
```

Comment:
I guess `getAsBoolean`?

Reply:
I'm indifferent on `getAsBoolean` or `asBoolean`. Both are better than `getBoolean`.

Reply:
never mind, I see this is addressed in the latest commit.